main/javascript-modifications/ses-guide.md

```diff
@@ -0,0 +1,505 @@
+This will provide a seamless experience as simple as <script src=“app.js”> or node app.js, but 
+with the safety of SES.
+  
+**tyg todo: My problem with this section is that while I think I understand what compartments are
+and what they do, I'm not clear on just how/when they are used. This (and the SES reference) could
+use what amounts to a paragraph or two like "When writing a contract, you use compartments like this. 
+This is when you want to create a compartment. This is when you don't want to. Do you ever destroy a
+compartment? If so, how?**
+
+
+## `lockdown()`
+
+`lockdown()` freezes all JavaScript defined objects accessible to any 
+program in the execution environment. Calling `lockdown()` turns a JavaScript
+system into a SES system, with enforced OCap (object-capability) security. It
+alters the surrounding execution environment, or realm, such that no two 
+programs running in the same realm can observe or interfere with each other 
+until they have been introduced.
+
+To do this, lockdown() tamper-proofs all of the JavaScript intrinsics to prevent 
+prototype pollution. After that, no program can subvert the methods of these objects 
+(preventing some man in the middle attacks). Also, no program can use these mutable
+objects to pass notes to parties that haven't been expressly introduced (preventing 
+some covert communication channels).
+
+For a full explanation of `lockdown()` and its options, please see
+[here](./lockdown.md).
+
+## `harden()`
+
+`harden()` should be called on all objects that will be transferred across a trust boundary 
+The general rule is if you make a new object and give it to someone else (and don't 
+immediately forget it yourself), you should give them `harden(obj)` instead of the raw object. 
+
+After calling lockdown, the harden function ensures that every object in the transitive 
+closure over property and prototype access starting with that object has been frozen by 
+`Object.freeze()`. This means that the object can be passed among programs and none of 
+those programs will be able to tamper with the surface of that object graph. They can 
+only read the surface data and call the surface functions.
+
+You have to harden a class before you harden any of its instances; i.e. it takes two separate 
+steps to harden both a class and its instances. Harden a base class before hardening classes 
+that inherit from it. harden()` does transitive freezing by following the object’s own
+properties (as opposed to properties it inherited), and the objects whose own properties refer 
+to them, and so forth.
+
+When you freeze an object via `harden()`, it ensures any external callers 
+can only interact with it through functions in the object’s API. `harden()` 
+is an enhanced transitive version of `Object.freeze()`, which only locks up an 
+object's own properties.
+
+All objects that are transferred across a trust boundary must have their API
```

Comment:
Didn't we say this above, but with "should" rather than "must"?

Reply:
I’m not well-versed at making that distinction. “should” seems appropriate since it is possible to pass unfrozen objects over a trust boundary, but “must” seems appropriate since doing so would invalidate the notion of trust over the boundary.

Reply:
Previous instance changed from "should" to "must".